vslib/src/SwitchStateBaseMACsec.cpp

```diff
@@ -0,0 +1,839 @@
+    // And a flow never belongs to two MACsec Port
+    if (getACLTable(aclEntryIds.front(), aclTableId) != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR(
+            "Cannot find corresponding ACL table for the entry %s",
+            sai_serialize_object_id(aclEntryIds.front()).c_str());
+
+        return SAI_STATUS_FAILURE;
+    }
+
+    // Find MACsec SC
+    attr.id = SAI_MACSEC_SC_ATTR_FLOW_ID;
+    attr.value.oid = macsecFlowId;
+    std::vector<sai_object_id_t> macsecScIds;
+    findObjects(SAI_OBJECT_TYPE_MACSEC_SC, attr, macsecScIds);
+
+    // One MACsec SC will only belong to one MACsec flow
+    // Meanwhile one MACsec flow will just belong to one port
+    if (macsecScIds.empty())
+    {
+        SWSS_LOG_ERROR(
+            "Cannot find corresponding MACsec SC for the flow %s",
+            sai_serialize_object_id(macsecFlowId).c_str());
+
+        return SAI_STATUS_FAILURE;
+    }
+
+    attr.id = SAI_MACSEC_SC_ATTR_MACSEC_DIRECTION;
+
+    CHECK_STATUS(get(SAI_OBJECT_TYPE_MACSEC_SC, macsecScIds.front(), 1, &attr));
+
+    auto direction = attr.value.s32;
+
+    // Find port
+    attr.id = (direction == SAI_MACSEC_DIRECTION_EGRESS) ? SAI_PORT_ATTR_EGRESS_MACSEC_ACL : SAI_PORT_ATTR_INGRESS_MACSEC_ACL;
+    attr.value.oid = aclTableId;
+
+    std::vector<sai_object_id_t> port_ids;
+
+    findObjects(SAI_OBJECT_TYPE_PORT, attr, port_ids);
+
+    if (port_ids.size() != 1)
+    {
+        SWSS_LOG_ERROR(
+            "Expect one port to one ACL table %s, but got %zu",
+            sai_serialize_object_id(aclTableId).c_str(), port_ids.size());
+
+        return SAI_STATUS_FAILURE;
+    }
+
+    portId = port_ids.front();
+
+    return SAI_STATUS_SUCCESS;
+}
+
+std::shared_ptr<HostInterfaceInfo> SwitchStateBase::findHostInterfaceInfoByPort(
+        _In_ sai_object_id_t portId)
+{
+    SWSS_LOG_ENTER();
+
+    std::shared_ptr<HostInterfaceInfo> info;
+
+    for (auto &kvp : m_hostif_info_map)
+    {
+        if (kvp.second->m_portId == portId)
+        {
+            info = kvp.second;
+            break;
+        }
+    }
+
+    if (info == nullptr)
+    {
+        SWSS_LOG_ERROR("Cannot find corresponding port %s", sai_serialize_object_id(portId).c_str());
+    }
+
+    return info;
+}
+
+sai_status_t SwitchStateBase::loadMACsecAttrFromMACsecPort(
+    _In_ sai_object_id_t objectId,
+    _In_ uint32_t attrCount,
+    _In_ const sai_attribute_t *attrList,
+    _Out_ MACsecAttr &macsecAttr)
+{
+    SWSS_LOG_ENTER();
+
+    const sai_attribute_t *attr = nullptr;
+
+    SAI_METADATA_GET_ATTR_BY_ID(attr, SAI_MACSEC_PORT_ATTR_MACSEC_DIRECTION, attrCount, attrList);
+
+    macsecAttr.m_direction = attr->value.s32;
+
+    SAI_METADATA_GET_ATTR_BY_ID(attr, SAI_MACSEC_PORT_ATTR_PORT_ID, attrCount, attrList);
+
+    auto portId = attr->value.oid;
+
+    macsecAttr.m_info = findHostInterfaceInfoByPort(portId);
+
+    if (macsecAttr.m_info == nullptr)
+    {
+        // The fail log has been recordedin findHostInterfaceInfoByPort
```

Comment:
i think this message should be here insted of line 383, since this functions expects info to be not null, but findHostInterfaceInfoByPort just searches for it, and whether it found or not should not be error, please move  that message here and remove from findHostInterfaceInfoByPort,

Reply:
Agree